tests/pipelines/eks/awscli-cl2-load-with-addons.yaml

```diff
@@ -28,34 +34,119 @@ spec:
     taskRef:
       kind: Task
       name:  slack-notification
+  - name: create-cluster-service-role
+    params:
+    - name: stack-name
+      value: $(params.cluster-name)-service-role
+    - name: role-cfn-url
+      value: $(params.service-role-cfn-url)s
+    - name: role-name
+      value: $(params.cluster-name)-service-role
+    runAfter:
+    - slack-notification
+    taskRef:
+      kind: Task
+      name: awscli-eks-role-create
+  - name: create-cluster-node-role
+    params:
+    - name: stack-name
+      value: $(params.cluster-name)-node-role
+    - name: role-cfn-url
+      value: $(params.node-role-cfn-url)
+    - name: role-name
+      value: $(params.cluster-name)-node-role
+    runAfter:
+    - slack-notification
+    taskRef:
+      kind: Task
+      name: awscli-eks-role-create
+  - name: get-preferred-role-decider
+    runAfter:
+      - create-cluster-node-role
+      - create-cluster-service-role
+    params:
+      - name: cfn-service-role-arn
+        value: $(tasks.create-cluster-service-role.results.role-arn)
+      - name: cfn-node-role-arn
+        value: $(tasks.create-cluster-node-role.results.role-arn)
+      - name: param-host-cluster-node-role-arn
+        value: $(params.host-cluster-node-role-arn)
+      - name: param-service-role-arn
+        value: $(params.servicerole)
+    taskSpec:
+        results:
+          - name: service-role-arn
+            description: holds the preferred service role Arn
+          - name: node-role-arn
+            description: holds the preferred node role Arn
+        params:
+          - name: cfn-service-role-arn
+          - name: cfn-node-role-arn
+          - name: param-host-cluster-node-role-arn
+          - name: param-service-role-arn
+        steps:
+          - name: get-prefered-role
+            image: alpine
+            script: |
```

Comment:
Any reason why we prefer an embeded one over ref here?

Reply:
I am open to thoughts here..the reason why i kept it embedded is it basically contains some pipeline orchestration logic to decide on which ARN to use. Since this orchestration logic is strictly tied with pipelines and doesn't have any meaning independently i wanted to keep it coupled with the pipeline.

Reply:
My only concern here is we are duplicating it here and `tests/pipelines/eks/awscli-eks-cl2-load.yaml`. It may get out-of-sync.
@hakuna-matatah WDYT?

Reply:
Yeah, I'm inclined towards not having to duplicate this logic across pipelines. I would rather suggest to make it less coupling, read `service-role-name` as param instead of relying on `arn` and try to fetch the `role-arn` with in the task based on the name supplied [
here](https://github.com/awslabs/kubernetes-iteration-toolkit/pull/327/files#diff-566fabd5277b56d45d422840baf5a353ec32543ea2f755ad6aa45eca934830deR40-R43) 

Same for node-role-arn too.

Reply:
okay..I have updated the tasks and pipelines to accept role names and currently we have assumptions in the task for default role name based on the cluster role name.